spec.html

```diff
@@ -30057,62 +30091,68 @@
         <p>The production <emu-grammar>Atom :: PatternCharacter</emu-grammar> evaluates as follows:</p>
         <emu-alg>
           1. Let _ch_ be the character matched by |PatternCharacter|.
           1. Let _A_ be a one-element CharSet containing the character _ch_.
-          1. Call CharacterSetMatcher(_A_, *false*) and return its Matcher result.
+          1. Call CharacterSetMatcher(_A_, *false*, _direction_) and return its Matcher result.
         </emu-alg>
         <p>The production <emu-grammar>Atom :: `.`</emu-grammar> evaluates as follows:</p>
         <emu-alg>
           1. Let _A_ be the set of all characters except |LineTerminator|.
-          1. Call CharacterSetMatcher(_A_, *false*) and return its Matcher result.
+          1. Call CharacterSetMatcher(_A_, *false*, _direction_) and return its Matcher result.
         </emu-alg>
         <p>The production <emu-grammar>Atom :: `\` AtomEscape</emu-grammar> evaluates as follows:</p>
         <emu-alg>
-          1. Return the Matcher that is the result of evaluating |AtomEscape|.
+          1. Return the Matcher that is the result of evaluating |AtomEscape| with argument _direction_.
         </emu-alg>
         <p>The production <emu-grammar>Atom :: CharacterClass</emu-grammar> evaluates as follows:</p>
         <emu-alg>
           1. Evaluate |CharacterClass| to obtain a CharSet _A_ and a Boolean _invert_.
-          1. Call CharacterSetMatcher(_A_, _invert_) and return its Matcher result.
+          1. Call CharacterSetMatcher(_A_, _invert_, _direction_) and return its Matcher result.
         </emu-alg>
         <p>The production <emu-grammar>Atom :: `(` Disjunction `)`</emu-grammar> evaluates as follows:</p>
         <emu-alg>
-          1. Evaluate |Disjunction| to obtain a Matcher _m_.
+          1. Evaluate |Disjunction| with argument _direction_ to obtain a Matcher _m_.
           1. Let _parenIndex_ be the number of left-capturing parentheses in the entire regular expression that occur to the left of this |Atom|. This is the total number of <emu-grammar>Atom :: `(` Disjunction `)`</emu-grammar> Parse Nodes prior to or enclosing this |Atom|.
           1. Return an internal Matcher closure that takes two arguments, a State _x_ and a Continuation _c_, and performs the following steps:
             1. Let _d_ be an internal Continuation closure that takes one State argument _y_ and performs the following steps:
               1. Let _cap_ be a copy of _y_'s _captures_ List.
               1. Let _xe_ be _x_'s _endIndex_.
               1. Let _ye_ be _y_'s _endIndex_.
-              1. Let _s_ be a new List whose characters are the characters of _Input_ at indices _xe_ (inclusive) through _ye_ (exclusive).
+              1. If _direction_ is equal +1, then
+                1. Let _s_ be a fresh List whose characters are the characters of _Input_ at indices _xe_ (inclusive) through _ye_ (exclusive).
+              1. Else, let _direction_ is equal to -1.
```

Comment:
This wording is somewhat unusual. Maybe better wording would be, "

1. If _direction_ is equal **to** +1, then
     1. Let _s_ be a fresh List whose characters are the characters of _Input_ at indices _xe_ (inclusive) through _ye_ (exclusive).
1. Else,
    1. **Assert:** _direction_ is equal to -1
    1. Let _s_ be a fresh List whose characters are the characters of _Input_ at indices _ye_ (inclusive) through _xe_ (exclusive).